js/src/api/subscriptions/personal.js

```diff
@@ -37,12 +39,33 @@ export default class Personal {
     ]);
   }
 
+  // FIXME: Because of the different API instances, the "wait for valid changes" approach
+  // doesn't work. Since the defaultAccount is critical to operation, we poll in exactly
+  // same way we do in ../eth (ala same as eth_blockNumber) and update. This should be moved
+  // to pub-sub as it becomes available
   _defaultAccount = () => {
+    const nextTimeout = (timeout = 1000) => {
+      setTimeout(() => {
+        this._defaultAccount();
+      }, timeout);
+    };
+
+    if (!this._api.transport.isConnected) {
+      nextTimeout(500);
+      return;
+    }
+
     return this._api.parity
       .defaultAccount()
       .then((defaultAccount) => {
-        this._updateSubscriptions('parity_defaultAccount', null, defaultAccount);
-      });
+        if (this._lastDefaultAccount !== defaultAccount) {
+          this._lastDefaultAccount = defaultAccount;
+          this._updateSubscriptions('parity_defaultAccount', null, defaultAccount);
+        }
+
+        nextTimeout();
+      })
+      .catch(nextTimeout);
```

Comment:
The error will be the first argument of `nextTimeout` right ? So I guess `setTimeout` would break